xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp

```diff
@@ -509,7 +534,30 @@ void CRenderer::Notify(const Observable& obs, const ObservableMessage msg)
       m_forceUpdateOverlayStyle = true;
       break;
     }
+    case ObservableMessageSubtitleAlignChanged:
+    {
+      m_subtitleVerticalMargin = GetSubtitleVerticalMargin();
```

Comment:
Did you consider thread safety? Looks dangerous doing a whole lot of things here on Notify.

Reply:
right added

Reply:
It is good practice to keep processing in callbacks to a bare minimum. Deliver data, set a flag, but don't do any calculations, calling functions, etc. The locks might block the render thread too long.

Reply:
ok then i have replaced all with a single variable change